main.go

```diff
@@ -98,23 +99,49 @@ func main() {
 	log.Infof("Gatehering public group(s)")
 
 	var publicGroup []string
-	for _, groupName := range releaseOptions.GroupNames {
-		groupName = strings.TrimSpace(groupName)
-		if len(groupName) == 0 {
-			continue
+	if cfg.DistributeAllGroup {
+		log.Infof("Gatehering all public group(s)")
+
+		groups, err := appAPI.AllGroups()
+		if err != nil {
+			failf("Failed to fetch groups, error: %s", err)
 		}
 
-		log.Printf("- %s", groupName)
+		for _, group := range groups {
+			if err := releaseAPI.AddGroup(group); err != nil {
+				failf("Failed to add group(%s) to the release, error: %s", group.DisplayName, err)
+			}
 
-		group, err := appAPI.Groups(groupName)
+			if group.IsPublic {
+				publicGroup = append(publicGroup, group.DisplayName)
+			}
+		}
```

Comment:
nit: this is duplicated across both the if and else branches, would it be possible to extract it to a function and call that with the list of groups as the argument?